src/DotNetEnv/Configuration/EnvConfigurationProvider.cs

```diff
@@ -37,9 +38,16 @@ public override void Load()
                 }
             }
 
-            // Since the Load method does not take care of cloberring, We have to check it here!
+            // Since the Load method does not take care of clobberring, We have to check it here!
             var dictionaryOption = options.ClobberExistingVars ? CreateDictionaryOption.TakeLast : CreateDictionaryOption.TakeFirst;
-            foreach (var value in values.ToDotEnvDictionary(dictionaryOption))
+            var dotEnvDictionary = values.ToDotEnvDictionary(dictionaryOption);
+
+            if (!options.ClobberExistingVars)
+                foreach (string key in Environment.GetEnvironmentVariables().Keys)
```

Comment:
Can you explain the thought process for this? You already have all the values that are present in the .env file, and the keys for those. Why do you need the list of actual env vars (keys)?